### ISSUE TITLE
Rule proposal: prevent mistake in map access

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1663]
### Description

Prevent mistake when check existence with `.has()`, but `.get()` or `.set()` with another key, I believe this is mistake in most cases. Especially in ternary.

### Fail

```js
const foo = map.has(key) ? map.get(anotherKey) : value;
```

```js
if (!map.has(key)) {
	const foo = map.set(anotherKey);
}
```

```js
if (!map.has(key)) {
	map.set(anotherKey, value);
}
```

### Pass

```js
const foo = map.has(key) ? map.get(key) : value;
```

```js
if (map.has(key)) {
	map.set(key, value);
	map.set(anotherKey, value);
}
```

**Comment by sindresorhus** [comment]
Accepted
